### ISSUE TITLE
Is there a way to keep miniconda2 and miniconda3 from being loaded simultaneously

### ISSUE THREAD

**Issue by hartzell** [spack/spack#4071]
`miniconda[23]` provide overlapping commands and should not both be `module loaded` simultaneously.

I'd like to avoid user-foot-shooting and disallow this.

Lmod has a [conflict](http://lmod.readthedocs.io/en/latest/050_lua_modulefiles.html) command that might be useful, but I'm not sure how to get at it from the respective packages.

Spack's [`conflicts`](http://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=setup_environment#conflicts) seems to address other issues (build time) and I don't think that it's relevant.

Is there a standard way to express this relationship in a package and get it into the modulefiles?

Thanks!

**Comment by adamjstewart** [comment]
I believe the correct Lmod command is [family](http://lmod.readthedocs.io/en/latest/050_lua_modulefiles.html). @alalazo Is there a way to set the family from a `package.py`? I noticed that `llvm` already does this, although I'm not sure if it works. Common families would include 'compiler' and 'MPI'. We could also add a 'miniconda' family.

**Comment by hartzell** [comment]
It just occured to me to check how this works with Python.  My normal environment has `@2.7.13` loaded, so I did this:

```
$ module load python/3.6.0

The following have been reloaded with a version change:
  1) python/2.7.13 => python/3.6.0

$
```

It believe that it can tell that they're two different versions of the same thing because they have the same name, just different version numbers.

Sadly, there's miniconda2/4.3.11 and miniconda3/4.3.11 so that won't work.

Adding a family seems like a big hammer, but...

**Comment by adamjstewart** [comment]
Yes, Lmod prevents you from loading multiple versions of the same package, but it has no way of knowing whether or not `miniconda2` and `miniconda3` will conflict. So we can either combine `miniconda2` and `miniconda3` into a single package, or add a `family` directive. Since `family` is important for compilers and MPI implementations, we will need it anyway. Otherwise users can load OpenMPI and MVAPICH2 at the same time.

**Comment by hartzell** [comment]
I just tried adding `family = 'miniconda'` to miniconda[23]'s packages and generating modulefiles without seeing any difference.  Also tried `provides('miniconda')`.

I dug a bit deeper into it in my copy of @alalazo's 'module files from templates' branch and in the lmod template I can see that `family` statements are only emitted if the module file modifies `MODULEPATH` (e.g. for gcc and Core and ...).

**Comment by citibeth** [comment]
We could make `miniconda` be just one package instead of two.

> Sadly, there's miniconda2/4.3.11 and miniconda3/4.3.11 so that won't work.

We can version a little "creatively.  miniconda2/4.3.11 would turn into Spack `miniconda@2.4.3.11`, and miniconda3/4.3.11 would turn into Spack `miniconda@3.4.3.11`.

**Comment by becker33** [comment]
You can do this now (and as of several years ago) using `projections` to rename the packages to the same name.

```
projections:
  miniconda2: "miniconda/{version}"
  miniconda3: "miniconda/{version}"
  all: "{name}/{version}"
```
